src/openms/include/OpenMS/ANALYSIS/ID/FragmentIndex.h

```diff
@@ -132,10 +175,47 @@ namespace OpenMS
      */
     struct Fragment
     {
-      UInt32 peptide_idx; // TODO: check size in sage implementation (32bit vs 64bit)
+      UInt32 peptide_idx; // 32 bit in sage
       float fragment_mz;
     };
 
+
+    /**
+     * @brief queries exactly one peak, with a set range of potential peptides, isotope error and precursor charge Hits are transfered into the a PSM
+     * @param candidates
+     * @param peak
+     * @param candidates_range
+     * @param isotope_error
+     * @param precursor_charge
+     */
+    void queryPeak(SpectrumMatchesTopN& candidates,
+                   const Peak1D& peak,
+                   std::pair<size_t, size_t> candidates_range,
+                   int16_t isotope_error,
+                   uint16_t precursor_charge);
+
+    /**
+     * @brief queries all peaks of a spectrum. loops over all preset isotope errors. The precursor-mass-window is ignored
+     * @param spectrum experimental query-spectrum
+     * @param mz
+     * @param sms
+     * @param charge
+     */
+    void closedSearch(const MSSpectrum& spectrum, float mz, SpectrumMatchesTopN& sms, uint16_t charge);
```

Comment:
this function looks a bit redundant to the query peak one? can you clarify a bit in the doc why a separate one is needed? e.g., does the other one call this one?